ts/main.ts

```diff
@@ -251,9 +347,23 @@ $sidebarMenu.addEventListener('click', (event: Event): void => {
   const selectedView = $eventTarget.closest('a')?.getAttribute('id');
   if (!selectedView) throw new Error('selectedView not found.');
 
-  $sidebar.classList.toggle('hidden');
-  $main.classList.toggle('overlay');
+  $sidebar.classList.toggle('is-open');
+  $main.classList.toggle('disable-elements');
+  if (selectedView === 'favorites') {
+    for (let i = 0; i < data.favorites.length; i++) {
+      const $movieCard = renderCard(data.favorites[i]);
+      $favoritesRow.appendChild($movieCard);
+    }
+  }
   viewSwap(selectedView);
-})
+});
+
+$detailsIcon.addEventListener('click', (): void => {
+  if ($detailsIcon.classList.contains('fa-solid')) {
+    $detailsIcon.className = 'fa-regular fa-heart fa-2xl details-icon';
+  } else if ($detailsIcon.classList.contains('fa-regular')) {
+    $detailsIcon.className = 'fa-solid fa-heart fa-2xl details-icon';
```

Comment:
Shouldn't this icon also add it to favorites when it makes it solid? (You don't need to have the functionality to remove from favorites but it should still add)